src/app/zone-ingresses/data/index.spec.ts

```diff
@@ -0,0 +1,239 @@
+        expect(actual.zoneIngress.availableServices.length).toStrictEqual(0)
+      },
+    )
+
+    test(
+      '1 availableService',
+      async ({ fixture }) => {
+        const actual = await fixture.setup((item) => {
+          return item
+        }, {
+          env: {
+            KUMA_SERVICE_COUNT: '1',
+          },
+        })
+        expect(Array.isArray(actual.zoneIngress.availableServices)).toStrictEqual(true)
+        expect(actual.zoneIngress.availableServices.length).toStrictEqual(1)
+      },
+    )
+  })
+  describe('addresses', () => {
+    test(
+      'absent addresses, socketAddress = "" advertisedSocketAddress = ""',
+      async ({ fixture }) => {
+        const actual = await fixture.setup((item) => {
+          if (typeof item.zoneIngress.networking === 'undefined') {
+            item.zoneIngress.networking = {}
+          }
+          delete item.zoneIngress.networking?.address
+          item.zoneIngress.networking.port = '80'
+          delete item.zoneIngress.networking?.advertisedAddress
+          item.zoneIngress.networking.advertisedPort = '80'
+          return item
+        })
+        expect(actual.zoneIngress.socketAddress).toStrictEqual('')
+        expect(actual.zoneIngress.advertisedSocketAddress).toStrictEqual('')
+      },
+    )
+
+    test(
+      'absent ports, socketAddress = "" advertisedSocketAddress = ""',
+      async ({ fixture }) => {
+        const actual = await fixture.setup((item) => {
+          if (typeof item.zoneIngress.networking === 'undefined') {
+            item.zoneIngress.networking = {}
+          }
+          item.zoneIngress.networking.address = '127.0.0.1'
+          delete item.zoneIngress.networking.port
+          item.zoneIngress.networking.advertisedAddress = '127.0.0.1'
+          delete item.zoneIngress.networking.advertisedPort
+          return item
+        })
+        expect(actual.zoneIngress.socketAddress).toStrictEqual('')
+        expect(actual.zoneIngress.advertisedSocketAddress).toStrictEqual('')
+      },
+    )
+
+    test(
+      'numbered ports, socketAddress is correct advertisedSocketAddress is correct',
+      async ({ fixture }) => {
+        const actual = await fixture.setup((item) => {
+          if (typeof item.zoneIngress.networking === 'undefined') {
+            item.zoneIngress.networking = {}
+          }
+          item.zoneIngress.networking.address = '127.0.0.1'
+          // @ts-ignore
+          item.zoneIngress.networking.port = 80
+          item.zoneIngress.networking.advertisedAddress = '127.0.0.2'
+          // @ts-ignore
+          item.zoneIngress.networking.advertisedPort = 81
+          return item
+        })
+        expect(actual.zoneIngress.socketAddress).toStrictEqual('127.0.0.1:80')
+        expect(actual.zoneIngress.advertisedSocketAddress).toStrictEqual('127.0.0.2:81')
+      },
+    )
+
+    test(
+      'string ports, socketAddress is correct advertisedSocketAddress is correct',
```

Comment:
They are always strings aren't they? Sounds like its easy to get mixed up as to whether these are strings or numbers.

Reply:
I don’t think so, no. I think they’re never strings.

Reply:
![Screenshot 2023-12-15 at 15 13 55](https://github.com/kumahq/kuma-gui/assets/554604/b0291216-5726-40a8-b5c7-41a7b25a32ff)

?

Reply:
I don’t think that’s correct. I can’t find a single place in kuma where `"port": "` appears for example.

Reply:
🤷 https://github.com/kumahq/kuma-gui/commit/3a7b2d4a27db892241b7af74bb6ad31cbe169994#diff-d91c72b0c4408f78487da8710045c189275b3aee8df371b01b5ad358056f0255

Is this a case of our types being wrong again?

Reply:
I think the final outcome here should be that we alter the type, but would be good to check that all `port` properties are numbers, and and such I'd rather try not to change that here but instead in its own explicit follow up PR.

I was in two minds on the need for the test, but I think the final thing that made me decide "yeah we should keep these tests here for both strings and numbers" was how easy it was for us to mix these up. The tests here prove that it works, even if we make a mistake with the types.

Related: I think its even more important that we use JS template literals for concatenation of things that we treat as strings (even ports which use integers for 'identifiers') rather than mathematical addition.

Reply:
I made an issue for the about `port` thing

https://github.com/kumahq/kuma-gui/issues/2069